torch/_library/infer_schema.py

```diff
@@ -124,7 +124,13 @@ def unstringify_type(ty: Union[type[object], str]) -> tuple[typing.Any, bool]:
         annotation_type, _ = unstringify_type(param.annotation)
 
         if annotation_type not in SUPPORTED_PARAM_TYPES:
-            if annotation_type.__origin__ is tuple:
+            if annotation_type == torch._C.ScriptObject:
```

Comment:
I think we should be able to get this to work.
The idea is to provide some new NewType annotation that behaves like "ScriptObject" (or Any) during typing, but can be uniquely identified by this function.

https://mypy.readthedocs.io/en/stable/more_types.html#newtypes